test/plugins/tokens.test.js

```diff
@@ -0,0 +1,189 @@
+'use strict';
+
+const assert = require('chai').assert;
+const sinon = require('sinon');
+const hapi = require('hapi');
+const mockery = require('mockery');
+const urlLib = require('url');
+const hoek = require('hoek');
+const testToken = require('./data/token.json');
+
+sinon.assert.expose(assert, { prefix: '' });
+require('sinon-as-promised');
+
+const getTokenMock = (token) => {
+    const mock = hoek.clone(token);
+
+    mock.update = sinon.stub();
+    mock.regenerate = sinon.stub();
+    mock.toJson = sinon.stub().returns(token);
+    mock.remove = sinon.stub();
+
+    return mock;
+};
+
+const getUserMock = (user) => {
+    const mock = hoek.clone(user);
+
+    mock.tokens = sinon.stub();
+
+    return mock;
+};
+
+const tokensGetterMock = tokens => Promise.resolve(tokens);
+
+const generateProfile = sinon.stub();
+
+describe('token plugin test', () => {
+    const username = 'ifox';
+    const userId = testToken.userId;
+    let tokenFactoryMock;
+    let userFactoryMock;
+    let authMock;
```

Comment:
why we need the authMock here? This is only creating a token and save it in datastore. We shouldn't need to mock auth stuff.